lib/pos_blockchain.go

```diff
@@ -506,8 +509,19 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
 		if currentBlock.IsCommitted() {
 			return nil, RuleErrorDoesNotExtendCommittedTip
 		}
+		if currentBlock.IsValidateFailed() {
+			return nil, RuleErrorAncestorBlockValidationFailed
+		}
+		if currentBlock.Header.ProposedInView >= prevView {
+			return nil, RuleErrorAncestorHasGreaterOrEqualView
+		}
+		if uint64(currentBlock.Header.Height)+1 != prevHeight {
+			return nil, RuleErrorAncestorHeightNotEqualChildHeightMinusOne
```

Comment:
Maybe this is more descriptive of what's happening?
```suggestion
			return nil, RuleErrorParentBlockHeightNotSequentialWithChildBlockHeight
```